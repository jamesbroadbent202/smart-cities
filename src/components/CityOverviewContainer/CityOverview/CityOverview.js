import React from 'react';
import PropTypes from 'prop-types';
import PageWrapper from '../../PageWrapper/PageWrapper';
import OverviewSection from '../../OverviewSection/OverviewSection';

const CityOverview = props => (
  <PageWrapper cityId={props.city.id}>
    <OverviewSection city={props.city} />
  </PageWrapper>
);

CityOverview.propTypes = {
  city: PropTypes.shape({
    name: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
  }).isRequired,
};

export default CityOverview;
