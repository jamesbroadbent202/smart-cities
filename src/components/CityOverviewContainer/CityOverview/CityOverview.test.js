import React from 'react';
import { shallow } from 'enzyme';
import CityOverview from './CityOverview';

jest.mock('../../OverviewSection/OverviewSection', () => 'OverviewSection');

const defaultProps = {
  city: {
    name: 'Perth',
    id: 'perth',
  },
};

it('should match Snapshot', () => {
  const component = shallow(
    <CityOverview {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});
