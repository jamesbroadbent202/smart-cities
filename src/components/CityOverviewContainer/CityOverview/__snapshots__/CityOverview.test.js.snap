// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<PageWrapper cityId=\\"perth\\" categoryId=\\"overview\\">
  <OverviewSection city={{...}} />
</PageWrapper>"
`;
